src/vptranslator.rs

```diff
@@ -0,0 +1,253 @@
+// Copyright 2017 Kisio Digital and/or its affiliates.
+//
+// This program is free software: you can redistribute it and/or
+// modify it under the terms of the GNU General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see
+// <http://www.gnu.org/licenses/>.
+
+//! See function translate
+
+use crate::objects::{Date, ExceptionType, ValidityPeriod};
+use chrono::{Datelike, Duration, Weekday};
+use num_traits::cast::FromPrimitive;
+use std::collections::BTreeSet;
+use std::vec::Vec;
+
+///Indicates whether service is available on the date specified.
+#[derive(Debug)]
+pub struct ExceptionDate {
+    ///Date of exception
+    pub date: Date,
+    ///exception type
+    pub exception_type: ExceptionType,
+}
+
+impl PartialEq for ExceptionDate {
+    fn eq(&self, other: &ExceptionDate) -> bool {
+        self.exception_type == other.exception_type
+    }
+}
+
+///Presents a list of dates in the form of intervals and exception dates.
+#[derive(Default, Debug)]
+pub struct BlockPattern {
+    ///Indicates operating days of the service
+    pub operating_days: Vec<Weekday>,
+    ///Start and end service day for the service interval
+    pub validity_periods: Vec<ValidityPeriod>,
+    ///List of dates where service is available or not in interval.
+    pub exceptions: Vec<ExceptionDate>,
+}
+
+fn get_prev_monday(date: Date) -> Date {
+    date + Duration::days(-1 * date.weekday().num_days_from_monday() as i64)
```

Comment:
can't you do `date - ...` ?

Reply:
I had an anomaly and I thought it was coming from there and I forgot to put the code back.
I'm sorry